torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -278,6 +282,41 @@ void SimpleValue::fillInLoopInfo(
       it->output()->replaceAllUsesWith(cur_elem);
       it.destroyCurrent();
     }
+  } else if (val_type->isSubtypeOf(TensorType::get())) {
+    // fill in max_trip_count_val
+    Value* outermost_dim_index = g.insertConstant(0, IntType::get(), loc);
+    // zero-dim tensor error handling
+    Value* num_dim = g.insert(aten::dim, {val}, {}, loc);
+    Value* cond_value = g.insert(aten::eq, {num_dim, outermost_dim_index}, {}, loc);
+    Node* if_node = g.insertNode(g.create(prim::If, 0)->setSourceRange(loc));
+    if_node->addInput(cond_value);
+
+    Block* true_block = if_node->addBlock();
+    if_node->addBlock();
+    {
+      WithInsertPoint guard(true_block);
+      g.insert(prim::RaiseException,
+          {std::string("iteration over a 0-d tensor!")}, {}, loc);
+    }
+
+    Value* sizes_tuple = g.insert(aten::size, {val}, {}, loc);
+    Value* max_trip_count_val = g.insert(aten::select, {sizes_tuple, outermost_dim_index}, {}, loc);
+    n->insertInput(0, max_trip_count_val);
+    // fill in the target element assignment value in the beginning of the FOR loop
+    {
+      TORCH_CHECK(iters_size == 1, "more than one iterable for in tensor");
+      Block* body_block = n->blocks()[0];
+      // replace the first Placeholder node in the block with the correct assignment
+      auto it = body_block->nodes().begin();
```

Comment:
maybe we could pass a list of values (`prim::Placeholder`s) directly instead of size?